app/member-reminders/member-reminders-ui/src/main/kotlin/com/hedvig/android/memberreminders/ui/MemberReminderCards.kt

```diff
@@ -0,0 +1,228 @@
+package com.hedvig.android.memberreminders.ui
+
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.expandVertically
+import androidx.compose.animation.fadeIn
+import androidx.compose.animation.fadeOut
+import androidx.compose.animation.shrinkVertically
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.icons.Icons
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.hedvig.android.core.designsystem.component.button.HedvigContainedSmallButton
+import com.hedvig.android.core.designsystem.material3.onWarningContainer
+import com.hedvig.android.core.designsystem.material3.warningContainer
+import com.hedvig.android.core.designsystem.material3.warningElement
+import com.hedvig.android.core.designsystem.preview.HedvigPreview
+import com.hedvig.android.core.designsystem.theme.HedvigTheme
+import com.hedvig.android.core.icons.Hedvig
+import com.hedvig.android.core.icons.hedvig.normal.Warning
+import com.hedvig.android.core.ui.infocard.VectorInfoCard
+import com.hedvig.android.memberreminders.ApplicableMemberReminders
+import com.hedvig.android.memberreminders.UpcomingRenewal
+import com.hedvig.android.notification.permission.NotificationPermissionState
+import hedvig.resources.R
+import kotlinx.datetime.Clock
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.atStartOfDayIn
+import kotlinx.datetime.daysUntil
+import kotlinx.datetime.toLocalDateTime
+
+@Composable
+fun MemberReminderCards(
+  memberReminders: ApplicableMemberReminders,
+  navigateToConnectPayment: () -> Unit,
+  openUrl: (String) -> Unit,
+  notificationPermissionState: NotificationPermissionState,
+  snoozePermission: () -> Unit,
+  modifier: Modifier = Modifier,
+) {
+  Column(
+    modifier = modifier,
+    verticalArrangement = Arrangement.spacedBy(8.dp),
+  ) {
+    AnimatedVisibility(
```

Comment:
A MemberReminder could then contain the data needed to construct its own ReminderCard. Then we would only need to add new MemberReminders, we have a generic way of creating the cards.

Reply:
I was thinking that we're most likely not going to be adding many (if any) new reminders here, plus what I wrote in the comment above is what made me write it this way.
There still is a generic way to render a card, with the `MemberReminderCards` composable, which in order for it to be able to render any card, it needs to know how to handle the data needed + callbacks for them for all of them.